prow.sh

```diff
@@ -0,0 +1,916 @@
+# for the repo.
+#
+# The expected environment is:
+# - $GOPATH/src/<import path> for the repository that is to be tested,
+#   with PR branch merged (when testing a PR)
+# - running on linux-amd64
+# - bazel installed (when testing against Kubernetes master), must be recent
+#   enough for Kubernetes master
+# - kind (https://github.com/kubernetes-sigs/kind) installed
+# - optional: Go already installed
+
+# Sets the default value for a variable if not set already and logs the value.
+# Any variable set this way is usually something that a repo's .prow.sh
+# or the job can set.
+configvar () {
+    # Ignore: Word is of the form "A"B"C" (B indicated). Did you mean "ABC" or "A\"B\"C"?
+    # shellcheck disable=SC2140
+    eval : \$\{"$1":="\$2"\}
+    eval echo "\$3:" "$1=\${$1}"
+}
+
+# Go versions can be specified seperately for different tasks
+# If the pre-installed Go is missing or a different
+# version, the required version here will get installed
+# from https://golang.org/dl/.
+configvar CSI_PROW_GO_VERSION_BUILD 1.11.4 "Go version for building the component" # depends on component's source code
+configvar CSI_PROW_GO_VERSION_K8S 1.12.1 "Go version for building Kubernetes for the test cluster" # depends on Kubernetes version
+configvar CSI_PROW_GO_VERSION_E2E 1.12.1 "Go version for building the Kubernetes E2E test suite" # depends on CSI_PROW_E2E settings below
+configvar CSI_PROW_GO_VERSION_SANITY "${CSI_PROW_GO_VERSION_BUILD}" "Go version for building the csi-sanity test suite" # depends on CSI_PROW_SANITY settings below
+configvar CSI_PROW_GO_VERSION_KIND "${CSI_PROW_GO_VERSION_BUILD}" "Go version for building 'kind'" # depends on CSI_PROW_KIND_VERSION below
+configvar CSI_PROW_GO_VERSION_GINKGO "${CSI_PROW_GO_VERSION_BUILD}" "Go version for building ginkgo" # depends on CSI_PROW_GINKGO_VERSION below
+
+# kind version to use. If the pre-installed version is different,
+# the desired version is downloaded from https://github.com/kubernetes-sigs/kind/releases/download/
+# (if available), otherwise it is built from source.
+configvar CSI_PROW_KIND_VERSION 0.2.1 "kind"
+
+# ginkgo test runner version to use. If the pre-installed version is
+# different, the desired version is built from source.
+configvar CSI_PROW_GINKGO_VERSION v1.7.0 "Ginkgo"
+
+# Ginkgo runs the E2E test in parallel. The default is based on the number
+# of CPUs, but typically this can be set to something higher in the job.
+configvar CSI_PROW_GINKO_PARALLEL "-p" "Ginko parallelism parameter(s)"
+
+# Enables building the code in the repository. On by default, can be
+# disabled in jobs which only use pre-built components.
+configvar CSI_PROW_BUILD_JOB true "building code in repo enabled"
+
+# Kubernetes version to test against. This must be a version number
+# (like 1.13.3) for which there is a pre-built kind image (see
+# https://hub.docker.com/r/kindest/node/tags), "latest" (builds
+# Kubernetes from the master branch) or "release-x.yy" (builds
```

Comment:
Yes, for example the `ci-kubernetes-csi-1-14-on-kubernetes-1-13` job does that. Whether we want to cover that case is debatable. It can detect regressions in Kubernetes that are not detected in Kubernetes itself because the E2E testing there only uses a single deployment.

Reply:
Then we may need to change the GO version to match the Kubernetes supported go version for that release.  There have been bugs in the past caused by using new go versions to build older Kubernetes releases (because go broke backwards compatibility)

Reply:
Okay.